src/main/java/dev/openfeature/sdk/NoOpProvider.java

```diff
@@ -59,11 +59,22 @@ public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double default
 
     @Override
     public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue,
-        EvaluationContext invocationContext) {
+                                                         EvaluationContext invocationContext) {
         return ProviderEvaluation.<Value>builder()
                 .value(defaultValue)
                 .variant(PASSED_IN_DEFAULT)
                 .reason(Reason.DEFAULT.toString())
                 .build();
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (obj instanceof FeatureProviderWrapper) {
```

Comment:
This is needed to keep the current tests succeeding.  
Furthermore, a `FeatureProviderWrapper` should be equal to its `FeatureProvider`, as they are one functional unit, and should behave to the outside as if there was no wrapper. Consequently, it the `FeatureProviderWrapper` equals its `FeatureProvider`, the reverse should also be true.

Reply:
> Consequently, it the `FeatureProviderWrapper` equals its `FeatureProvider`, the reverse should also be true.

But we can't ensure this with this implementation strategy, because we do not know who is implementing the `FeatureProvider` interface. All implementations of `FeatureProvider` out there (that do not extend `NoopProvider` or `EventProvider` are not covered by this approach and will behave differently.